### ISSUE TITLE
Warnings with custom_lint on Dart 2.18.0

### ISSUE THREAD

**Issue by kuhnroyal** [invertase/github-action-dart-analyzer#133]
Not exactly sure this is the right place to post this.

Staring with Dart 2.18.0 or Flutter 3.3.0 the action starts showing new warnings coming from the custom_lint package.
This is weird since my understanding still is that analyzer plugins are not hooked into `dart analyze`, maybe something has changed there.

I am suddenly seeing this warning in the logs: https://github.com/invertase/dart_custom_lint/blob/main/packages/custom_lint/lib/src/analyzer_plugin/lints.dart#L71

```
Run invertase/github-action-dart-analyzer@cdd8652b05bf7ed08ffce30f425436780f869f13
> Dart Analyzer - Infos
> Dart Analyzer - Warnings
  /home/github/actions-runner-1/_work/my_package/example/pubspec.yaml:16:2 The plugin is currently starting
> Dart Analyzer - Errors
0 analyzer hints detected.
1 analyzer warnings detected.
0 analyzer errors detected.
Error: Dart Analyzer detected problems, see the logs for more information.
```

**Comment by kuhnroyal** [comment]
FYI @rrousselGit Maybe this issue better fits in the custom_lint repo, not sure.

**Comment by rrousselGit** [comment]
This definitely should be transfered to custom_lint

It's weird that this happens though

**Comment by kuhnroyal** [comment]
> It's weird that this happens though

Yea that's what I thought. But I can reproduce this locally on Flutter 3.3.0.

@Salakar Can you please transfer the issue?
